salttesting/helpers.py

```diff
@@ -1131,10 +1136,10 @@ def kill_children(_children, terminate=False, kill=False):
                 except psutil.NoSuchProcess:
                     _children.remove(child)
 
-        kill_children(children)
+        kill_children([child for child in children if child.is_running() and not any(sys.argv[0] in cmd for cmd in child.cmdline())])
```

Comment:
This is confusing me a little bit. I don't see exception handling for this condition elsewhere but I do see some handling for cases where this is `None`. I added a fallback for OSX for the time being though.